ios/RNCAsyncStorageDelegate.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^RNCAsyncStorageCompletion)(NSError *_Nullable error);
+typedef void (^RNCAsyncStorageResultCallback)(NSArray<id<NSObject>> * valuesOrErrors);
+
+@protocol RNCAsyncStorageDelegate <NSObject>
+
+- (void)allKeys:(RNCAsyncStorageResultCallback)block;
+
+- (void)mergeValues:(NSArray<NSString *> *)values
+            forKeys:(NSArray<NSString *> *)keys
+         completion:(RNCAsyncStorageResultCallback)block;
+
+- (void)removeAllValues:(RNCAsyncStorageCompletion)block;
+
+- (void)removeValuesForKeys:(NSArray<NSString *> *)keys
+                 completion:(RNCAsyncStorageResultCallback)block;
+
+- (void)setValues:(NSArray<NSString *> *)values
+          forKeys:(NSArray<NSString *> *)keys
+       completion:(RNCAsyncStorageResultCallback)block;
+
+- (void)valuesForKeys:(NSArray<NSString *> *)keys
+           completion:(RNCAsyncStorageResultCallback)block;
+
+@optional
+
+@property (nonatomic, readonly, getter=isPassthrough) BOOL passthrough;
```

Comment:
Please follow Camel-Case naming.

Reply:
Not quite sure what you mean. [Passthrough](https://en.wikipedia.org/wiki/Passthrough) is one word.

Reply:
Ahha, It's a word, sorry, but seems react-native uses `passThrough`, https://github.com/facebook/react-native/search?q=passThroughTouches&unscoped_q=passThroughTouches.

Reply:
Yeah, it's a bit confusing but I think it's verb vs. noun. `passThroughTouches` returns whether touches should be passed down to child components, whilst here we're returning whether the delegate should be treated as a passthrough.